### ISSUE TITLE
Update ImGUI to latest v1.62

### ISSUE THREAD

**Issue by Rominitch** [SaschaWillems/Vulkan#496]
Dear Devs,

I created a fork on my account with new ImGUI 1.62.
I enabled DEPRECATED features into imconfig.h and clean all code.

My main issue is: I disabled ini file because imGUI vertices will be to 0 (I report bug to ocornut too).

Don't hesitate to merge into master branch !

**Comment by SaschaWillems** [comment]
Thx, I'll take a look at it. My spare time is very limited right now, so it may take a while.

**Comment by Rominitch** [comment]
After spoke with ocornut, I adapt my code to something better than my first commit.

See https://github.com/ocornut/imgui/issues/1893

Regards !

**Comment by Warpten** [comment]
A couple things I changed to try to get the sample triangle to work with imgui 1.62:

- Added `ImGui::CreateContext()` before the call to `glfwInit()`

At that point, it looked ok, but then all of a sudden `void UIOverlay::update()` has `imDrawData->TotalVtxCount = 0` and `imDrawData->TotalIdxCount = 0`. Fair enough, I said, and added

```cpp
    if (vertexBufferSize == 0 || indexBufferSize == 0)
        return;
```

Then, in `VulkanExampleBase::submitFrame()`, all of a sudden `&UIOverlay->cmdBuffers[currentBuffer];` throws a subscript out of range exception, because `cmdBuffers.size() == 0`. I "fixed" it by preventing queue submission if `cmdBuffers.size() == 0`, but now I get errors in the validation layer since it leads to unsignalled/unwaited semaphores in queues.

**Comment by SaschaWillems** [comment]
I have just merged the ImGui 1.6x branch into master. Along with updating to a recent ImGui version I also reworked how the UI overlay works, getting rid of a lot of code by making it part of the example command buffer. Changes were tested with windows, linux and android, so all platforms should be fine.
